aws/saml.go

```diff
@@ -113,6 +113,9 @@ func ExtractRoleArnAndPrincipalArn(samlResponse SAMLResponse) (string, string, e
 			s := strings.Split(v.Value, ",")
 			roleArn := s[0]
 			principalArn := s[1]
+			if roleName != "" && !strings.HasSuffix(roleArn, roleName) {
+				continue
+			}
```

Comment:
`HasSuffix` can cause a problem. For example, when there are two roles "SuperTestRole" and "TestRole" and "TestRole" is specified as `roleName`, `strings.HasSuffix("SuperTestRole", "TestRole")` can return true. This is not expected behavior.

I suggest to split `roleArn` by "/" and compare the last substring with `roleName`.

```suggestion
			splitStrings := strings.Split(roleArn, "/")
			attrRoleName := splitStrings[len(splitStrings) - 1]
			if roleName != "" && attrRoleName != roleName {
				continue
			}
```

Reply:
I overlooked that.
Fixed 8d256af.